mungegithub/misc-mungers/deployment/kubernetes/configmap.yaml

```diff
@@ -5,7 +5,7 @@ organization: kubernetes
 project: kubernetes
 # Run blunderbuss before approval-handler, so that we can suggest approvers
 # based on assigned reviewer.
```

Comment:
How does the new approval-handler interact with blunderbuss? Is this comment still relevant?

Reply:
Somewhat. Both the new and old approval mechanisms consider the current assignees when determining who to suggest as an approver. So if the approve plugin happens to run before the blunderbuss munger it **might** suggest a different (but still valid) approver than it would have otherwise suggested. When the approval notification is eventually updated it will suggest the person that it would have suggested originally if there is a difference.
Once #5547 is merged and deployed to k/k, blunderbuss won't assign users, it will instead request reviews from them so there will no longer be any interaction between blunderbuss and approvals.

Reply:
I saw #5547, but I'm not sure review requests are necessarily the best way to go about things, at least in the immediate term (per my comment on the issue https://github.com/kubernetes/test-infra/issues/4179#issuecomment-345419948).

How does the current assignees impact approver selection? Is it worth having that logic at all?

Reply:
Any current assignees that contribute to the approval of the PR (i.e. can approve at least one file) are included in the suggested approvers. Additional users that are suggested as approvers must approve a file that is not approved by a current assignee. https://github.com/kubernetes/test-infra/blob/master/prow/plugins/approve/approvers/owners.go#L448
This logic isn't mandatory, it is just intended to direct PR authors to get approval from assignees first if possible. If the approve plugin runs before the blunderbuss munger assigns **user** we might not include **user** in the suggested approvers initially, but the users that are suggested will still be valid and when the notification is updated it will include **user**.
We could get rid of the assignee logic in the approve plugin, but it would make the suggestions strictly worse as they would be more likely to suggest approvers that are unnecessary assuming all assignees eventually give approval.